exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSystemSchemaFactory.java

```diff
@@ -52,9 +55,20 @@
 
   private List<WorkspaceSchemaFactory> factories;
   private String schemaName;
+  protected FileSystemPlugin plugin;
 
   public FileSystemSchemaFactory(String schemaName, List<WorkspaceSchemaFactory> factories) {
     super();
+    if (factories.size() > 0 ) {
+      this.plugin = factories.get(0).getPlugin();
```

Comment:
A comment would be helpful: what is special about plugin 0 that it should be the default? Is plugin 0 the default of some sort?